pkg/util/log/log.go

```diff
@@ -394,6 +412,20 @@ func Error(v ...interface{}) error {
 	return err
 }
 
+// ErrorStackDepth logs at the error level and the given stack depth and returns an error containing the formated log message
+func ErrorStackDepth(depth int, v ...interface{}) error {
```

Comment:
in terms of interface, I think the `depth` should be an _additional_ depth that's added on top of whatever depth is needed for `DatadogLogger` itself, since the caller of this method shouldn't have to know what initial depth is needed for this method of `DatadogLogger` to work well.